hbase-server/src/main/java/org/apache/hadoop/hbase/util/RegionMover.java

```diff
@@ -435,6 +451,10 @@ public boolean unload() throws InterruptedException, ExecutionException, Timeout
           LOG.debug("List of region servers: {}", regionServers);
           return false;
         }
+        // Remove RS present not in the designated file
+        if (designatedFile != null) {
+          filterDesignatedServers(regionServers);
```

Comment:
I haven't thought of how to elegantly combine excludeFile and designedFile in one method. Could you please give some advice?

Reply:
We can combine `filterDesignatedServers()` and `stripExcludes()`.
If you see carefully, both are doing the same thing except for one is excluding what is present in file and another is including. So whether to include/exclude all RS present in given file is something you can provide as boolean argument and act accordingly.
Something like:
```
private void includeExcludeRegionServers(String fileName, List<ServerName> regionServers, boolean isInclude)
```

For, filterDesignatedServers(), it can be:
```
includeExcludeRegionServers(designatedFile, onlineServers, true);
```
and stripExcludes can be replaced with:
```
includeExcludeRegionServers(excludeFile, regionServers, false);
```

Sounds good?

Reply:
Done as you said. Thanks for your advice.